src/moneyTracker.js

```diff
@@ -0,0 +1,25 @@
+function Moneytracker() {
+  this._transactions = [];
+  this._totalBalance = 0;
+};
+
+Moneytracker.prototype.recordEarning = function (money, date) {
+  this._transactions.push(new Transaction(money, date))
+};
+
+Moneytracker.prototype.recordExpenditure = function (money, date) {
+  this._transactions.push(new Transaction(money, date))
+};
+
+Moneytracker.prototype._updateTotalBalance = function () {
+  for(var i = 0; i < this._transactions.length; i++) {
+    this._totalBalance += this._transactions[i].credit + this._transactions[i].debit;
+    this._transactions[i].balance = this._totalBalance;
+  };
+};
+
+Moneytracker.prototype.printStatement = function () {
```

Comment:
Running `printStatement` changes the data on the transactions.  It's generally best to avoid side effects if they're not necessary because they make code have more possible states it can get into.  Eg the state of the transactions is different before and after `printStatement` is run (because each transaction's `balance` attribute has been set).

You could avoid this by making the `Statement` object calculate but not store the balance when it's printed.